tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -1543,5 +1543,124 @@ int main(int argc, char **argv)
+     *#     indicated cipher suite.
+     **/
+    {
+        DEFER_CLEANUP(struct s2n_cert_chain_and_key *chain_and_key,
+                      s2n_cert_chain_and_key_ptr_free);
+        EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
+                                                       S2N_DEFAULT_ECDSA_TEST_CERT_CHAIN,
+                                                       S2N_DEFAULT_ECDSA_TEST_PRIVATE_KEY));
+
+        DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(),
+                      s2n_config_ptr_free);
+        EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, chain_and_key));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(config, "default_tls13"));
+        EXPECT_SUCCESS(s2n_config_disable_x509_verification(config));
+        EXPECT_SUCCESS(s2n_setup_ticket_key(config));
+
+        DEFER_CLEANUP(struct s2n_connection *server_conn = s2n_connection_new(S2N_SERVER),
+                      s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(server_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+        DEFER_CLEANUP(struct s2n_connection *client_conn = s2n_connection_new(S2N_CLIENT),
+                      s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(client_conn);
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+
+        struct s2n_test_io_pair io_pair = { 0 };
+        EXPECT_SUCCESS(s2n_io_pair_init_non_blocking(&io_pair));
+        EXPECT_SUCCESS(s2n_connections_set_io_pair(client_conn, server_conn, &io_pair));
+
+        /* Force the HRR path */
+        client_conn->security_policy_override = &security_policy_test_tls13_retry;
+
+        client_conn->actual_protocol_version = S2N_TLS13;
+        client_conn->secure.cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+
+        /* Set a resumption psk */
+        DEFER_CLEANUP(struct s2n_stuffer psk_identity = { 0 }, s2n_stuffer_free);
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&psk_identity, 0));
+        EXPECT_OK(s2n_setup_encrypted_ticket(client_conn, &psk_identity));
+        struct s2n_offered_psk_list identity_list = { .conn = client_conn };
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&identity_list.wire_data, 0));
+        EXPECT_OK(s2n_write_test_identity(&identity_list.wire_data, &psk_identity.blob));
+        EXPECT_OK(s2n_select_resumption_psk(server_conn, &identity_list));
+
+        struct s2n_psk_parameters *psk_params = &client_conn->psk_params;
+        struct s2n_psk *psk = psk_params->chosen_psk;
+
+        /* Add arbitrary amount to ticket age so a nonzero value implies it's been processed */
+        client_conn->psk_params.chosen_psk->ticket_age_add = 10;
+
+        /* ClientHello 1 */
+        EXPECT_SUCCESS(s2n_client_hello_send(client_conn));
+
+        /* Read the obfuscated ticket age from ClientHello 1 */
+        client_conn->handshake.io.read_cursor = 342;
```

Comment:
So, we usually don't alter stuffer variables directly, as it's kind of unsafe. We use something like this, where the length to skip is a const that describes where we're skipping to:
`EXPECT_SUCCESS(s2n_stuffer_skip_read(hello_stuffer, LENGTH_TO_SESSION_ID));`

I would go through and use stuffer methods anywhere you're altering stuffers in this test.

Reply:
Okay yeah that makes sense. I fixed those in 365037c and ff5b6f2.